grid/social-media/upload-url/profile.supr

```diff
@@ -0,0 +1,222 @@
+
+"""
+Upload Assets via URL to Social Media sites
+Upload images and videos available via an URL, onto social media such as Instagram, Facebook, or Twitter.
+"""
+
+name = "social-media/upload-url"
+version = "1.0.0"
+
+"""
+Upload an Asset
+Register an upload of an asset to a social media profile, in preparation for publishing later.
+"""
+usecase RegisterUpload unsafe {
+  input {
+    """
+    Profile ID
+    Identifier of a profile to upload to. May be optional with some providers.
+    """
+    profileId string!
+
+    """
+    Caption
+    Caption of the asset
+    """
+    caption string!
+
+    """
+    URL
+    URL of the asset. Must be publically acessible.
+    """
+    url! string!
+
+    """
+    Type
+    Media type of the asset
+    """
+    type enum {
```

Comment:
This is risky: you are adding "reel" as enum, but this is specific just to Instagram. Al's type is very overloaded term. Consider changing this to a string and renaming it to provider Type. Alternatively make this optional and add new required field like "mimeType" from which we can infer a sane default.

Reply:
YouTube Shorts are pretty much 1:1 to Reels and doesn't map to "provider type" in the same way it does for Instagram

Reply:
yes

Reply:
The mapping is map's concern, my thought was that you'd provide input like `providerType: "short"` or `providerType: "reel"`. But maybe we could adopt some universal name like `short boolean`?

Reply:
not `short boolean` because that implies the video is *short* (whereas it also needs to be vertical), perhaps `reel boolean`, but then what about images?